File: java/reverse-string/src/main/java/ReverseString.java
class ReverseString {

    String reverse(String inputString) {
        byte[] b = inputString.getBytes();
        for (int i = 0; i < b.length / 2; i++) {
            byte tmp = b[i];
            b[i] = b[b.length - i - 1];
            b[b.length - i - 1] = tmp ;
        }

        return new String(b);
    }
  
}

File: go/reverse-string/reverse_string.go
package reverse

func Reverse(str string) string {
	if len(str) == 0 {
		return ""
	}

	var ret []rune
	rs := []rune(str)
	for i := len(rs) - 1; i >= 0; i -= 1 {
		ret = append(ret, rs[i])
	}

	return string(ret)
}
